smart_contracts/xgov_registry/deploy_config.py

```diff
@@ -222,18 +236,31 @@ def _set_roles() -> None:
     )
 
     algorand_client = AlgorandClient.from_environment()
-    deployer = algorand_client.account.from_environment("DEPLOYER")
+
+    # Try to create Vault signer first, fallback to environment if not available
+    try:
+        vault_signer = create_vault_signer_from_env()
+        deployer_address = vault_signer.address
+        logger.info(f"Using Vault transaction signer with address: {deployer_address}")
```

Comment:
Why are we repeating this step? Is that because Vault needs to repeat this for every single signature?